server/http_tunnel.go

```diff
@@ -31,22 +32,36 @@ func StartHTTPSTunnel(addr, externalProxy, certFile, KeyFile string) {
 		}}
 		proxy.ConnectDial = proxy.NewConnectDialToProxy(externalProxy)
 	}
+
+	proxy.OnRequest().HandleConnectFunc(func(host string, ctx *goproxy.ProxyCtx) (*goproxy.ConnectAction, string) {
+		log.Printf("HTTP Tunnel: got CONNECT request %s\n", host)
+		return nil, host
+	})
+
 	proxy.OnRequest().DoFunc(
 		func(r *http.Request, ctx *goproxy.ProxyCtx) (*http.Request, *http.Response) {
 			// Leave the request untouched.  Log a message for debugging purposes.
 			log.Printf("HTTP Tunnel: got request %s\n", r.URL)
 			return r, nil
 		})
 
-	if certFile == "" || KeyFile == "" {
+	if username != "" || password != "" {
+		auth.ProxyBasic(proxy, "tunnel", func(user, passwd string) bool {
+			authorized := (user == username && passwd == password)
+			log.Printf("HTTP Tunnel: user=%s, password=%s, authorized=%t\n", user, passwd, authorized)
```

Comment:
we're not really logging usernames and passwords right?  are these cleartext?

Reply:
We're logging the username and passwords that people send us.  It is cleartext once it gets to us.

Reply:
They're encrypted during transit by TLS.

Reply:
we shouldn't log usernames and passwords.

Reply:
Why not?

Reply:
Because usernames and passwords are sensitive customer data and we don't want to be responsible for leaking it.

Beyond writing these to a log file, why are we manually comparing usernames and passwords.  Does the library not handle that?

Reply:
@moonchen Here's an example of [a very high profile company that had to issue a public statement](https://blog.twitter.com/official/en_us/topics/company/2018/keeping-your-account-secure.html) when it was discovered they were writing customer passwords to a log file.  The statement goes on to request that all users reset their passwords.  We don't want to be responsible for a leak like this.

> Due to a bug, passwords were written to an internal log before completing the hashing process. We found this error ourselves, removed the passwords, and are implementing plans to prevent this bug from happening again.

Reply:
The correct username and password are available on the command line, but I see the possibility that the user may accidentally use a incorrect but sensitive password in an unsuccessful attempt.  I'll remove the info from the log.

Reply:
> Because usernames and passwords are sensitive customer data and we don't want to be responsible for leaking it.
> 
> Beyond writing these to a log file, why are we manually comparing usernames and passwords. Does the library not handle that?

The library doesn't handle the comparison for us.

Reply:
Just my 2 cents on password logging.  Even if the password is viewable because it's in the clear as a command line arg, doesn't mean that it should be clear in the log.  For example, there's nothing stopping our users from forwarding these logs via journald off to some log aggregator(ex: Pelion Cloud).  In that case, we've leaked their password out of the box.  Users wouldn't expect this because the general rule is that you never log secrets of any kind including passwords.